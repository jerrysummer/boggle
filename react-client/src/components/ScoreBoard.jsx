import React from 'react';

const ScoreBoard = (props) => {
  const wordToScore = (word) => {
    const len = word.length || 0;
    if (len <= 2) return 0;
    else if (len <= 4) return 1;
    else if (len === 5) return 2;
    else if (len === 6) return 3;
    else if (len === 7) return 5;
    else if (len >= 8) return 11;
  }; // convert string length to score

  const uniqueSelection = [...new Set(props.pastSelection)]; // removes dup strings from array; game rule

  const wordNScore = uniqueSelection.map(string => [string, wordToScore(string)]); // convert array of chars to array of arrays [char, score]

  let totalScore = 0;

  wordNScore.forEach((element) => {
    totalScore += element[1];
  }); // calculate total score using the new array that includes score

  return (
    <div className="scoreBoard">
      <table >
        <tbody>
          <tr>
            <th className="leftCol">Word</th>
            <th>Score</th>
          </tr>
          {
            wordNScore.map((arr, i) => (
              <tr key={i}>
                <td className="leftCol">{arr[0].toLowerCase()}</td>
                <td>{arr[1]}</td>
              </tr>
            ))
          }
          <tr>
            <th className="leftCol total">Total</th>
            <th className="total">{totalScore}</th>
          </tr>

        </tbody>
      </table>
    </div>
  );
};

export default ScoreBoard;
