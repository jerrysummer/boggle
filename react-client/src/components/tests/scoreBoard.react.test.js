import React from 'react';
import renderer from 'react-test-renderer';
import ScoreBoard from '../ScoreBoard';

it('renders correctly', () => {
  const component = renderer
    .create(<ScoreBoard pastSelection="['TEST','TEST','QUE','AAAAAAAAA','A']" />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
