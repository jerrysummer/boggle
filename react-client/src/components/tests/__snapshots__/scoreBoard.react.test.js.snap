// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="scoreBoard"
>
  <table>
    <tbody>
      <tr>
        <th
          className="leftCol"
        >
          Word
        </th>
        <th>
          Score
        </th>
      </tr>
      <tr>
        <td
          className="leftCol"
        >
          [
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="leftCol"
        >
          '
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="leftCol"
        >
          t
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="leftCol"
        >
          e
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="leftCol"
        >
          s
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="leftCol"
        >
          ,
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="leftCol"
        >
          q
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="leftCol"
        >
          u
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="leftCol"
        >
          a
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="leftCol"
        >
          ]
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <th
          className="leftCol total"
        >
          Total
        </th>
        <th
          className="total"
        >
          0
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
